modules/controllers/jobs-controller.ts

```diff
@@ -63,27 +134,77 @@ export function JobsController(tracer?: any) {
             if (!vaultAddress) {
                 throw new Error(`Chain not configured: ${chain}`);
             }
-            const viemClient = getViemClient(chain);
 
-            const blockNumber = await viemClient.getBlockNumber();
+            const fromBlock = (
+                await prisma.prismaPoolDynamicData.findFirst({
+                    orderBy: { blockNumber: 'desc' },
+                })
+            )?.blockNumber;
+
+            // Sepolia vault deployment block, uncomment to test from the beginning
+            // const fromBlock = 5274748n;
+
+            // Guard against unsynced pools
+            if (!fromBlock) {
+                throw new Error(`No synced pools found for chain: ${chain}`);
+            }
+
+            const pools = await prisma.prismaPool.findMany();
+            const dbIds = pools.map((pool) => pool.id.toLowerCase());
+            const viemClient = getViemClient(chain);
 
-            const changedPools = await getChangedPools(vaultAddress, viemClient, blockNumber, chain);
-            if (changedPools.length === 0) {
+            const { changedPools, latestBlock } = await getChangedPools(vaultAddress, viemClient, BigInt(fromBlock));
+            const ids = changedPools.filter((id) => dbIds.includes(id.toLowerCase())); // only sync pools that are in the database
+            if (ids.length === 0 || !latestBlock) {
                 return [];
             }
+            return syncPools(ids, viemClient, vaultAddress, chain, latestBlock + 1n);
+        },
+        /**
+         * Updates database pools with the subgraph and onchain state.
+         * Usually used only when database schema changes and new data needs to be added to the pools.
+         * Most likely used only together with migrations.
+         * Alternatively, it can be run as a new pools job, because it can be slow then.
+         *
+         * @param chainId
+         */
+        async updatePools(chainId: string) {
```

Comment:
isnt this the same as "reloadPools" above?

Reply:
true, removing